bundles/org.openhab.automation.pwm/README.md

```diff
@@ -0,0 +1,62 @@
+# Pulse Width Modulation (PWM) Automation
+
+This automation module implements [Pulse Width Modulation (PWM)](https://en.wikipedia.org/wiki/Pulse-width_modulation).
+
+PWM can be used to control actuators continuously from 0 to 100% that only support ON/OFF commands. E.g. valves or heating burners.
+It accomplishes that by switching the actuator on and off with a fixed interval.
+The higher the control percentage (duty cycle), the longer the ON phase.
+
+Example: If the fixed interval is 10 sec and the duty cycle is 30%, the output is ON for 3 sec and OFF for 7 sec.
+
+This module is **unsuitable** for controlling LED lights as the high PWM frequency can't be met.
+
+> Note: The module starts to work only if the duty cycle has been updated at least once.
+
+## Modules
+
+The PWM module can be used in openHAB's [rule engine](https://www.openhab.org/docs/configuration/rules-dsl.html). This automation provides a trigger and an action module.
+
+The trigger module has an input Item `dutycycleItem` (0-100%).
+It calculates the ON/OFF state and writes it to the action module, which sends the ON/OFF command the output Item `outputItem`.
```

Comment:
The action module does not seem to do anything besides passing the command to a specified item, correct?
I would hence claim that it isn't needed, because the very same can be achieved with the existing ItemCommandAction. This takes a `command` as an input from a trigger module, so all that the PWM trigger would have to do is to pass the ON/OFF command as an output with the id `command`.

Reply:
I just tried that but it fails [here](https://github.com/openhab/openhab-core/blob/79edf2b9e643ad802c31b2e5f54edb93534277fd/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemCommandActionHandler.java#L110). `inputs` contains a key `1.command`. Although my trigger handler sends `command`. It fails because ItemCommandActionHandler expects `command`.

I pushed the failing code. Am I sending the command wrongly?

Reply:
Ok, it took me a while, but I found the solution. The rule engine will actually only map inputs+outputs if they have tags - it is not done based on the input/output names (my thinking above was incorrect).
I have created https://github.com/openhab/openhab-core/pull/2378 to add tags to the core actions. If you now add a "command" tag to your output of the trigger module [like this](https://github.com/openhab/openhab-addons/pull/10205/files#r637609937), it all works as expected!

Reply:
Thanks you very much! I'll give it a try and change it in the PID controller, too.

Reply:
Works as intended! That simplified the code pretty nice.